wagtailinventory/migrations/0002_pageblock_unique_constraint.py

```diff
@@ -0,0 +1,39 @@
+from django.db import migrations, models
+from django.db.models import Min, Count
+
+
+def remove_duplicates(apps, schema_editor):
+    PageBlock = apps.get_model('wagtailinventory', 'PageBlock')
+
+    duplicate_pks_to_keep = (
+        PageBlock.objects
+            .values('page', 'block')
+            .annotate(Min('pk'), count=Count('pk'))
+            .filter(count__gt=1)
+            .values_list('pk__min', flat=True)
+    )
+
+    duplicates_to_keep = PageBlock.objects.in_bulk(duplicate_pks_to_keep)
+
+    for duplicate_to_keep in duplicates_to_keep.values():
+        duplicates_to_delete = PageBlock.objects.filter(
+            page=duplicate_to_keep.page,
+            block=duplicate_to_keep.block
+        ).exclude(pk=duplicate_to_keep.pk)
+
+        duplicates_to_delete.delete()
```

Comment:
Could the double-query here (to get dups to keep and use that to filter dups to delete) be replaced by a single query to get the dup with the larger pk? In some quick testing locally, this appears to get the pks of the same objects that duplicates_to_delete gets in the `for` loop.

```python
PageBlock.objects
    .values('page', 'block')
    .annotate(Max('pk'), count=Count('pk'))
    .filter(count__gt=1)
    .values_list('pk__max', flat=True)
```

I don't think it really matters too much either way in a migration. It might be good to have a comment or docstring here to explain what's going on though.

Reply:
Discussed this in chat.